infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategy.java

```diff
@@ -82,55 +83,63 @@ public CommonPVCStrategy(
       @Named("che.infra.kubernetes.pvc.access_mode") String pvcAccessMode,
       @Named("che.infra.kubernetes.pvc.precreate_subpaths") boolean preCreateDirs,
       PVCSubPathHelper pvcSubPathHelper,
-      KubernetesNamespaceFactory factory) {
+      KubernetesNamespaceFactory factory,
+      EphemeralWorkspaceAdapter ephemeralWorkspaceAdapter) {
     this.pvcName = pvcName;
     this.pvcQuantity = pvcQuantity;
     this.pvcAccessMode = pvcAccessMode;
     this.preCreateDirs = preCreateDirs;
     this.pvcSubPathHelper = pvcSubPathHelper;
     this.factory = factory;
+    this.ephemeralWorkspaceAdapter = ephemeralWorkspaceAdapter;
   }
 
   @Override
   public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
       throws InfrastructureException {
     final String workspaceId = identity.getWorkspaceId();
-    final Set<String> subPaths = new HashSet<>();
-    final PersistentVolumeClaim pvc = newPVC(pvcName, pvcAccessMode, pvcQuantity);
-    k8sEnv.getPersistentVolumeClaims().put(pvcName, pvc);
-    for (Pod pod : k8sEnv.getPods().values()) {
-      PodSpec podSpec = pod.getSpec();
-      for (Container container : podSpec.getContainers()) {
-        String machineName = Names.machineName(pod, container);
-        InternalMachineConfig machineConfig = k8sEnv.getMachines().get(machineName);
-        addMachineVolumes(workspaceId, subPaths, pod, container, machineConfig.getVolumes());
+    if (ephemeralWorkspaceAdapter.isEphemeral(workspaceId)) {
```

Comment:
Changing code like that shows a big diff I would say. Can you just:
```java
if (ephemeralWorkspaceAdapter.isEphemeral(workspaceId)) {
    ephemeralWorkspaceAdapter.provision(k8sEnv, identity);
    return;
}
```
?

Reply:
yeah, sure

Reply:
fixed